.github/workflows/prod.yml

```diff
@@ -0,0 +1,48 @@
+name: deploy to prod
+
+on:
+  release:
+    types: [created]
+  push:
+    branches:
+      - master
```

Comment:
Does this not trigger this workflow when a PR is merged to master? Or is this just the syntax that Github actions has instead of something like the following:

```
on:
    push:
        tags:
            - v*
```

Reply:
Good catch. You're right. That will run when pushed to master or a tag is pushed. On the other hand we don't want it to run when a tag is pushed to any branch other than master.

Reply:
Is there a way to do some sort of "and" logic here?  I've been combing the github actions documentation and haven't found anything.  Right now it seems like we are running on 

(release created) OR (push to master OR tag pushed)

What we want is 

(release created) OR (push to master AND tag pushed)

or ideally just:

(tag pushed to master)

Reply:
Based on the documentation on the [create workflow trigger](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#create), it looks like github actions are triggered using the [webhook for create](https://docs.github.com/en/developers/webhooks-and-events/webhook-events-and-payloads#create), which will have the information in the "ref_type" key/value pair whether it is a branch or a tag that is created.  Is it possible to leverage that from github actions as a trigger?  Not sure.

Reply:
YML syntax doesn't like mixing types in dicts, so this, which to me would mean AND of "push to master" and "tag like v*", doesn't work:

    on:
      push:
        branches:
          - master
          tags:
            - v*

Reply:
I think @loganballard is on to something w/using refs. Isn't there a path using refs that means "tags on master"?

Or maybe we want to use release tags as the filter?